panic: send on closed channel

#### Expected behavior

I shouldn't get a `panic` that kills the whole microservice.

#### Actual behavior

I got a `panic`.

#### Steps to reproduce

Have a lot of consumers with `pulsar.Exclusive` read from several topics and in another shell keep resetting cursors while the cluster is under heavy load. Lots of `[Failed to create consumer]` and `"[Failed to connect to broker.]"` are reported. At that point, chances are that you could get this panic:

```
panic: send on closed channel
goroutine 354 [running]:
github.com/apache/pulsar-client-go/pulsar/internal.(*connection).SendRequestNoWait(0xc0007123c0, 0xc000b38000, 0xc001584a00, 0xc000b38000)
	/go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/internal/connection.go:565 +0x85
github.com/apache/pulsar-client-go/pulsar/internal.(*rpcClient).RequestOnCnxNoWait(0xc00010f080, 0x1068a48, 0xc0007123c0, 0xa, 0x105cb00, 0xc001584a00, 0xc000b54ea0, 0xc0bbffdf9e3fceb2)
	/go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/internal/rpc_client.go:169 +0x83
github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).discardCorruptedMessage(0xc0005f6a80, 0xc001626040, 0x2)
	/go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:1103 +0x2d7
github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).MessageReceived(0xc0005f6a80, 0xc001626000, 0x106b6d8, 0xc001ba2000, 0xbb8e01, 0xc000712430)
	/go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:477 +0x1445
github.com/apache/pulsar-client-go/pulsar/internal.(*connection).handleMessage(0xc0007123c0, 0xc001626000, 0x106b6d8, 0xc001ba2000)
	/go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/internal/connection.go:639 +0x115
github.com/apache/pulsar-client-go/pulsar/internal.(*connection).internalReceivedCommand(0xc0007123c0, 0xc00039ec40, 0x106b6d8, 0xc001ba2000)
	/go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/internal/connection.go:528 +0x269
github.com/apache/pulsar-client-go/pulsar/internal.(*connection).run(0xc0007123c0)
	/go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/internal/connection.go:382 +0x31a
github.com/apache/pulsar-client-go/pulsar/internal.(*connection).start.func1(0xc0007123c0)
	/go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/internal/connection.go:215 +0x85
created by github.com/apache/pulsar-client-go/pulsar/internal.(*connection).start
	/go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/internal/connection.go:211 +0x3f
```

I also noticed that right before the `panic` I also got these two errors from the client:

```
"message": "[received too big frame size=1953068645 maxFrameSize=5253120]"
```

```
message: "[Discarding corrupted message]"
```

#### System configuration
**Pulsar client**: v0.5.0
**Pulsar version**: `streamnative/sn-pulsar:2.7.1.4`

--- a/pulsar/internal/connection.go
+++ b/pulsar/internal/connection.go
@@ -549,10 +549,15 @@ func (c *connection) SendRequest(requestID uint64, req *pb.BaseCommand,
 	if c.getState() == connectionClosed {
 		callback(req, ErrConnectionClosed)
 	} else {
-		c.incomingRequestsCh <- &request{
+		select {
+		case <-c.closeCh:
+			callback(req, ErrConnectionClosed)
+
+		case c.incomingRequestsCh <- &request{
 			id:       &requestID,
 			cmd:      req,
 			callback: callback,
+		}:
 		}
 	}
 }
@@ -562,12 +567,17 @@ func (c *connection) SendRequestNoWait(req *pb.BaseCommand) error {
 		return ErrConnectionClosed
 	}
 
-	c.incomingRequestsCh <- &request{
+	select {
+	case <-c.closeCh:
+		return ErrConnectionClosed
+
+	case c.incomingRequestsCh <- &request{
 		id:       nil,
 		cmd:      req,
 		callback: nil,
+	}:
+		return nil
 	}
-	return nil
 }
 
 func (c *connection) internalSendRequest(req *request) {

